apps/worker/src/app/workflow/usecases/run-job/run-job.usecase.ts

```diff
@@ -114,19 +119,54 @@ export class RunJob {
     }
   }
 
+  /*
+   * If the following condition is met,
+   * - transactions were merged to the main delayed digest.
+   * - the main delayed digest was canceled.
+   * that mean that we need to assign a new active digest follower job to replace it.
+   * so from now on we will continue the follower transaction as main digest job.
+   */
+  private assignNewDigestExecutor(activeDigestFollower: JobEntity): JobEntity {
+    return activeDigestFollower;
+  }
+
+  private isCanceledMainDigest(type: StepTypeEnum | undefined, status: JobStatusEnum) {
+    return type === StepTypeEnum.DIGEST && status === JobStatusEnum.CANCELED;
+  }
+
   @Instrument()
-  private async delayedEventIsCanceled(job: JobEntity): Promise<boolean> {
+  private async delayedEventIsCanceled(
+    job: JobEntity
+  ): Promise<{ canceled: boolean; activeDigestFollower: JobEntity | null }> {
+    let activeDigestFollower: JobEntity | null = null;
+
     if (job.type !== StepTypeEnum.DIGEST && job.type !== StepTypeEnum.DELAY) {
-      return false;
+      return { canceled: false, activeDigestFollower };
     }
 
-    const count = await this.jobRepository.count({
+    const canceled = job.status === JobStatusEnum.CANCELED;
+
+    if (job.status === JobStatusEnum.CANCELED) {
+      activeDigestFollower = await this.activeDigestMainFollowerExist(job);
+    }
+
+    return { canceled, activeDigestFollower };
+  }
+
+  @Instrument()
+  private async activeDigestMainFollowerExist(job: JobEntity): Promise<JobEntity | null> {
+    if (job.type !== StepTypeEnum.DIGEST) {
+      return null;
+    }
+
+    return await this.jobRepository.findOne({
```

Comment:
Does anyone with digest expertise could let me know if we need to fetch by digest key here?

Reply:
I do not think so

Reply:
@davidsoderberg shouldn't we? 
we do merge digests by the "digest key" so the logic that calculates the new "follower" in the cancel use-case is fine... however, I think that here we need to fetch by "digest key" because there might be many delayed digests with different "digest keys"...

Reply:
Good catch, my thought was that there should only be one delayed digest job for the specific subscriber and template so digest key would be not needed. But maybe it is good to add it as well to be 100% sure.

Reply:
Added 🙏

Reply:
Thanks for the feedback, I added digest key🙏